src/charm.py

```diff
@@ -90,11 +94,22 @@ def _render_conf_file(self) -> str:
             template = Template(file.read())
 
         rendered = template.render(
-            kratos_session_url=self._kratos_session_url,
-            kratos_login_url=self._kratos_login_url,
+            kratos_session_url=self._get_kratos_endpoint_info("sessions_endpoint"),
+            kratos_login_url=self._get_kratos_endpoint_info("login_browser_endpoint"),
         )
         return rendered
 
+    def _get_kratos_endpoint_info(self, key: str) -> Optional[str]:
+        if self.model.relations[self._kratos_relation_name]:
```

Comment:
A little Nitpick:

- Generally, we could return early when handling errors or unexpected data, like

```python
if not self.model.relations[self._kratos_relation_name]:
    logger.info("blah blah")
    return

try:
    what you wanna do
except SomeException:
    do what you need to do to handle it
```